teps/0110-decouple-catalog-organization-and-reference.md

```diff
@@ -0,0 +1,239 @@
+resources from the Catalog. This creates a hard coupling between the way we organize resources in the Tekton Catalog
+and the way user consume it. This coupling is forcing us into one possible Catalog organization and prevent us to make
+drastic changes to this organization without breaking users.
+
+This enhancement aims to decouple the catalog organization from users consumption of those resources. This enhancement
+proposal is mainly touching the Catalog and the Hub components. This would also push the Hub and Remote Resolution
+to be the main entrypoint for users to search, list and get Tekton resources.
+
+### Goals
+
+- Allow maintainers of `tektoncd/catalog` to change the organization without affecting the consumption of the resources.
+- Define a set of standard, supported ways to get a Tekton resource.
+
+### Non-Goals
+
+- Modify the current organization of the `tektoncd/catalog` repository.
+
+### Requirements
+
+- An end-user should never refer / install task from a URL that points directly to the GitHub repository. For example,
+  they should not use `https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildpacks/0.4/buildpacks.yaml`.
+
+## Proposal
+
+### Tekton Bundles
+
+Tekton publishes OCI images of resources from the Catalog using [Tekton Bundles][bundles]. Users can fetch these
+resources via the OCI image references: `gcr.io/tekton-releases/catalog/upstream/{name}:{version}`. For example, the
+*buildpacks* `Task` can be fetched from: `gcr.io/tekton-releases/catalog/upstream/buildpacks:0.4`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: buildpacks
+spec:
+  taskRef:
+    name: buildpacks
+    bundle: gcr.io/tekton-releases/catalog/upstream/buildpacks:0.4
+```
+
+### Tekton Hub
+
+#### HTTP Endpoint
+
+We propose that the Tekton Hub hosts HTTP endpoint that users can rely on to fetch Tekton resources. The Tekton Hub
+HTTP endpoint can take this format: `hub.tekton.dev/{catalog}/{kind}/{name}/{version}`.
+
+The Tekton Hub already hosts this endpoint: `https://api.hub.tekton.dev/v1/resource/{catalog}/{kind}/{name}/{version}`.
+For example, `https://api.hub.tekton.dev/v1/resource/Tekton/task/buildpacks/0.4` hosts the *buildpacks* `Task`. But the
+structure of the data there requires users to process the data first before applying the resources to their clusters.
+The new HTTP endpoint mentioned above should serve the *yaml* files that users can easily install resources to their
+clusters, similarly the current experience using GitHub HTTP endpoint:
+
+```shell
+# Current: using GitHub path in existing Tekton Catalog organization
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildpacks/0.4/buildpacks.yaml
+
+# Proposal: using the HTTP Endpoint from Tekton Catalog - should serve same yaml as the one above
+kubectl apply -f https://hub.tekton.dev/tekton//task/buildpacks/0.4/
```

Comment:
We have endpoint like https://api.hub.tekton.dev/v1/resource/Tekton/task/buildpacks/0.4/yaml to serve the yaml

Reply:
@piyush-garg yes, but it's not the yaml directly applied (it's encapsulated in a json payload)